src/main/java/pl/cleankod/exchange/core/usecase/FindAccountAndConvertCurrencyUseCase.java

```diff
@@ -34,7 +34,7 @@ public Optional<Account> execute(Account.Number number, Currency targetCurrency)
 
     private Money convert(Money money, Currency targetCurrency) {
         if (!baseCurrency.equals(targetCurrency)) {
-            return money.convert(currencyConversionService, targetCurrency);
+            return currencyConversionService.convert(money, targetCurrency);
```

Comment:
Use case class is a place where we would like to use some low-level components to orchestrate the business process, all the domain logic of how the currency is converted should be moved somewhere else.